tasks/sealsupra/supra_config.go

```diff
@@ -0,0 +1,297 @@
+package sealsupra
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type SystemInfo struct {
+	ProcessorCount int
+	CoreCount      int
+	ThreadsPerCore int
+	CoresPerL3     int
+}
+
+type CoordinatorConfig struct {
+	Core    int
+	Hashers int
+}
+
+type SectorConfig struct {
+	Sectors      int
+	Coordinators []CoordinatorConfig
+}
+
+type TopologyConfig struct {
+	PC1Writer       int
+	PC1Reader       int
+	PC1Orchestrator int
+	PC2Reader       int
+	PC2Hasher       int
+	PC2HasherCPU    int
+	PC2Writer       int
+	PC2WriterCores  int
+	C1Reader        int
+	SectorConfigs   []SectorConfig
+}
+
+type SupraSealConfig struct {
+	NVMeDevices []string
+	Topology    TopologyConfig
+}
+
+func GetSystemInfo() (*SystemInfo, error) {
+	cmd := exec.Command("hwloc-ls")
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("error running hwloc-ls: %v", err)
+	}
+
+	info := &SystemInfo{}
+	scanner := bufio.NewScanner(strings.NewReader(string(output)))
+
+	l3Regex := regexp.MustCompile(`L3 L#(\d+)`)
+	puRegex := regexp.MustCompile(`PU L#(\d+)`)
+	coreRegex := regexp.MustCompile(`Core L#(\d+)`)
+	packageRegex := regexp.MustCompile(`Package L#(\d+)`)
+
+	var currentL3Cores int
+	var lastL3Index int = -1
+	var threadCount int
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if l3Match := l3Regex.FindStringSubmatch(line); l3Match != nil {
+			l3Index, _ := strconv.Atoi(l3Match[1])
+			if lastL3Index != -1 && l3Index != lastL3Index {
+				if info.CoresPerL3 == 0 || currentL3Cores < info.CoresPerL3 {
+					info.CoresPerL3 = currentL3Cores
+				}
+				currentL3Cores = 0
+			}
+			lastL3Index = l3Index
+		}
+
+		if coreRegex.MatchString(line) {
+			info.CoreCount++
+			currentL3Cores++
+		}
+
+		if puRegex.MatchString(line) {
+			threadCount++
+		}
+
+		if packageRegex.MatchString(line) {
+			info.ProcessorCount++
+		}
+	}
+
+	// Handle the last L3 cache
+	if info.CoresPerL3 == 0 || currentL3Cores < info.CoresPerL3 {
+		info.CoresPerL3 = currentL3Cores
+	}
+
+	if info.CoreCount > 0 {
+		info.ThreadsPerCore = threadCount / info.CoreCount
+	}
+
+	return info, nil
+}
+
+func GenerateSupraSealConfig(info SystemInfo, dualHashers bool, batchSize int, nvmeDevices []string) (SupraSealConfig, error) {
```

Comment:
This rhymes with the CLI command (but could get out of sync in the future). If that's correct, having this be called from there may be better.